ui-components/src/helpers/tableSummaries.tsx

```diff
@@ -1,71 +1,91 @@
+  rentAsPercentIncomeRangeMax: string | number,
+  totalAvailable: number | undefined,
+  totalCount: number | undefined,
+  unitTypeName: string
+) => {
+  const minIncome =
+    minIncomeRangeMin == minIncomeRangeMax ? (
+      <strong>{minIncomeRangeMin}</strong>
+    ) : (
+      <>
+        <strong>{minIncomeRangeMin}</strong> {t("t.to")} <strong>{minIncomeRangeMax}</strong>
+      </>
+    )
 
-    const getRent = (rentMin: string, rentMax: string, percent = false) => {
-      const unit = percent ? `% ${t("t.income")}` : ` ${t("t.perMonth")}`
-      return rentMin == rentMax ? (
-        <>
-          <strong>{rentMin}</strong>
-          {unit}
-        </>
-      ) : (
-        <>
-          <strong>{rentMin}</strong> {t("t.to")} <strong>{rentMax}</strong>
-          {unit}
-        </>
-      )
-    }
+  const getRent = (rentMin: string | undefined, rentMax: string | undefined, percent = false) => {
+    const unit = percent ? `% ${t("t.income")}` : ` ${t("t.perMonth")}`
+    return rentMin == rentMax ? (
+      <>
+        <strong>{rentMin}</strong>
+        {unit}
+      </>
+    ) : (
+      <>
+        <strong>{rentMin}</strong> {t("t.to")} <strong>{rentMax}</strong>
+        {unit}
+      </>
+    )
+  }
 
-    // Use rent as percent income if available, otherwise use exact rent
-    const rent = unitSummary.rentAsPercentIncomeRange.min
-      ? getRent(
-          unitSummary.rentAsPercentIncomeRange.min.toString(),
-          unitSummary.rentAsPercentIncomeRange.max.toString(),
-          true
-        )
-      : getRent(unitSummary.rentRange.min, unitSummary.rentRange.max)
+  // Use rent as percent income if available, otherwise use exact rent
+  const rent = rentAsPercentIncomeRangeMin
+    ? getRent(rentAsPercentIncomeRangeMin.toString(), rentAsPercentIncomeRangeMax.toString(), true)
+    : getRent(rentRangeMin, rentRangeMax)
 
-    return {
-      unitType: <strong>{t(`listings.unitTypes.${unitSummary.unitType.name}`)}</strong>,
-      minimumIncome: (
+  return {
+    unitType: <strong>{t(`listings.unitTypes.${unitTypeName}`)}</strong>,
+    minimumIncome:
+      minIncomeRangeMin != undefined ? (
         <>
           {minIncome} {t("t.perMonth")}
         </>
+      ) : (
+        <strong>Data not available</strong>
       ),
-      rent: <>{rent}</>,
-      availability: (
-        <>
-          {unitSummary.totalAvailable > 0 ? (
-            <>
-              <strong>{unitSummary.totalAvailable}</strong>{" "}
-              {unitSummary.totalAvailable == 1 ? t("t.unit") : t("t.units")}
-            </>
-          ) : (
-            <span className="uppercase">{t("listings.waitlist.label")}</span>
-          )}
-        </>
-      ),
-      totalCount: (
-        <>
-          <strong>{unitSummary.totalCount}</strong>{" "}
-          {unitSummary.totalCount == 1 ? t("t.unit") : t("t.units")}
-        </>
-      ),
-    }
+    rent: <>{rentRangeMin != undefined ? rent : <strong>Data not available</strong>}</>,
```

Comment:
Reasoning about rent is a little tricky. (I think I see what's going on - we're checking if `rentRangeMin` is undefined because that's what used to calculate `rent`, but I think this could be made easier to read.) Could you change the code above to make it so that when we get here, `rent` is either a valid string that we should display, or `undefined`. (That way, we can check if `rent` is undefined here, instead of checking if `rentRangeMin` is undefined.)

Reply:
updated